clients/hadoopfs/src/test/java/io/lakefs/committer/LakeFSOutputCommitterTest.java

```diff
@@ -0,0 +1,163 @@
+
+        s3Bucket = "example";
+        s3Base = String.format("s3://%s", s3Bucket);
+        CreateBucketRequest cbr = new CreateBucketRequest(s3Bucket);
+        s3Client.createBucket(cbr);
+
+        SparkConf sparkConf = new SparkConf().setMaster("local").setAppName("Spark test")
+                .set("spark.sql.shuffle.partitions", "17")
+                .set("spark.hadoop.mapred.output.committer.class","io.lakefs.committer.DummyOutputCommitter")
+                .set("spark.hadoop.fs.lakefs.impl","io.lakefs.LakeFSFileSystem")
+                .set("fs.lakefs.access.key", "AKIAIOSFODNN7EXAMPLE")
+                .set("fs.lakefs.secret.key", "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY")
+                .set("fs.lakefs.endpoint", "http://localhost:8000/api/v1")
+                .set("fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+                .set(org.apache.hadoop.fs.s3a.Constants.ACCESS_KEY, S3_ACCESS_KEY_ID)
+                .set(org.apache.hadoop.fs.s3a.Constants.SECRET_KEY, S3_SECRET_ACCESS_KEY)
+                .set(org.apache.hadoop.fs.s3a.Constants.ENDPOINT, s3Endpoint);
+        this.spark = SparkSession.builder().config(sparkConf).getOrCreate();
+    }
+
+    @Ignore
+    @Test
+    public void testWriteTextfile() throws ApiException, IOException {
+        String output = "lakefs://example/main/some_df";
+
+        StructType structType = new StructType().add("A", DataTypes.StringType, false);
+
+        List<Row> nums = ImmutableList.of(
+                RowFactory.create("value1"),
+                RowFactory.create("value2"));
+        Dataset<Row> df = spark.createDataFrame(nums, structType);
+
+        ArrayList<String> failed = new ArrayList<String>();
+
+        // TODO(Lynn): Add relevant file formats
+        List<String> fileFormats = Arrays.asList("text");
+        for (String fileFormat : fileFormats) {
+            try {
+                // save data in selected format
+                String outputPath = String.format("%s.%s", output, fileFormat);
+                logger.info(String.format("writing file in format %s to path %s", fileFormat, outputPath));
+                df.write().format(fileFormat).save(outputPath);
+
+                // read the data we just wrote
+                logger.info(String.format("reading file in format %s", fileFormat));
+                Dataset<Row> df_read = spark.read().format(fileFormat).load(outputPath);
+                Dataset<Row> diff = df.exceptAll(df_read);
+                Assert.assertTrue("all rows written were read", diff.isEmpty());
+                diff = df_read.exceptAll(df);
+                Assert.assertTrue("all rows read were written", diff.isEmpty());
+
+            } catch (Exception e) {
+                logger.error(String.format("Format %s unexpected exception: %s", fileFormat, e));
+                failed.add(fileFormat);
+            }
+        }
+
+        Assert.assertEquals("Failed list is not empty: %s" , 0, failed.size());
```

Comment:
Using assertThat with an appropriate matcher, as [suggested here](https://stackoverflow.com/a/51279675/192263), might be both more readable _and_ give a better error message!